zarr/tests/test_n5.py

```diff
@@ -35,3 +39,17 @@ def test_partial_chunk_decode(chunk_shape: Tuple[int, ...]):
     chunk[subslices] = 1
     subchunk = np.ascontiguousarray(chunk[subslices])
     assert np.array_equal(codec_wrapped.decode(codec_wrapped.encode(subchunk)), chunk)
+
+
+def test_dtype_decode():
+    store = 'data/array.n5'
+    n5_store = N5FSStore(store)
+    z = zarr.open(n5_store)
+    z.create_dataset("test", shape=100, dtype="u1")
+    dtype_n5 = json.loads(n5_store["test/.zarray"])["dtype"]
+    shutil.rmtree(store)
```

Comment:
consider using `atexit.register` here instead, e.g. https://github.com/zarr-developers/zarr-python/blob/4dc6f1f5046708648b5e6b82d0a64a24d1a40566/zarr/tests/test_storage.py#L992

Reply:
Thank you! i am using it now